### PR TITLE
Add package resolution for @types/react

### PR DESCRIPTION
**Pull request by jhaals**
Add resolution for version 17 `@types/react` due to breaking changes introduced in version 18.

### CODE REVIEW

File: .changeset/little-moles-pull.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+'@backstage/create-app': patch
+---
+
+Add type resolutions for `@types/react` and `types/react-dom`. This is an
```
Comment (freben) [inline .changeset/little-moles-pull.md:5]: missing words

File: .changeset/little-moles-pull.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+To apply these changes to your existing installation. Add a resolutions block to your `package.json`
```
Comment (freben) [inline .changeset/little-moles-pull.md:7]:
```suggestion
To apply these changes to your existing installation, add a resolutions block to your `package.json`
```

File: .changeset/little-moles-pull.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+```json
+  "resolutions": {
+    "@types/react": "^17",
+    "@types/react-dom": "^17"
+  },
+```
+
+If your existing app depend on react 16 use this resolution block instead.
```
Comment (freben) [inline .changeset/little-moles-pull.md:16]:
```suggestion
If your existing app depends on react 16, use this resolution block instead.
```

**Review by freben — APPROVED** [review]
preapproved but some spelling

**Review by Rugvip — APPROVED** [review]

**Pull request merged**